todo-list/src/hooks/todos/useTodo.ts

```diff
@@ -0,0 +1,54 @@
+import { useMutation, useQuery } from "react-query";
+import { addTodos, fetchTodosByUserId } from "../../api/todo";
+import { Todo } from "../../types/todos/todos.type";
+
+export function useTodos() {
+  const userId = localStorage.getItem("userId");
+  const queryResult = useQuery<Todo[], Error>(
+    ["todos", userId],
+    () => fetchTodosByUserId(Number(userId)),
+    {
+      enabled: !!userId,
+      retry: 3,

```

Comment:
`retry` should accompanied with `retryDelay` to avoid hitting backend with multiple request at the same time

https://tanstack.com/query/v4/docs/framework/react/reference/useQuery